flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/TimeFunctionsITCase.java

```diff
@@ -732,6 +775,46 @@ private Stream<TestSetSpec> floorTestCases() {
                                 $("f2").floor(TimeIntervalUnit.MILLENNIUM),
                                 "FLOOR(f2 TO MILLENNIUM)",
                                 LocalDateTime.of(2001, 1, 1, 0, 0),
-                                TIMESTAMP().nullable()));
+                                TIMESTAMP().nullable())
+                        .testResult(
+                                $("f2").cast(TIMESTAMP_LTZ(3))
+                                        .floor(TimeIntervalUnit.SECOND)
+                                        .cast(STRING()),
+                                "CAST(FLOOR(CAST(f2 AS TIMESTAMP_LTZ(3)) TO SECOND) AS STRING)",
+                                LocalDateTime.of(2020, 2, 29, 1, 56, 59, 0)
+                                        .format(
+                                                DateTimeFormatter.ofPattern(
+                                                        "yyyy-MM-dd' 'HH:mm:ss.SSS")),
+                                STRING().nullable())
+                        .testResult(
+                                $("f2").cast(TIMESTAMP_LTZ(3))
+                                        .floor(TimeIntervalUnit.MINUTE)
+                                        .cast(STRING()),
+                                "CAST(FLOOR(CAST(f2 AS TIMESTAMP_LTZ(3)) TO MINUTE) AS STRING)",
+                                LocalDateTime.of(2020, 2, 29, 1, 56, 0, 0)
+                                        .format(
+                                                DateTimeFormatter.ofPattern(
+                                                        "yyyy-MM-dd' 'HH:mm:ss.SSS")),
+                                STRING().nullable())
+                        .testResult(
+                                $("f2").cast(TIMESTAMP_LTZ(3))
+                                        .floor(TimeIntervalUnit.HOUR)
+                                        .cast(STRING()),
+                                "CAST(FLOOR(CAST(f2 AS TIMESTAMP_LTZ(3)) TO HOUR) AS STRING)",
+                                LocalDateTime.of(2020, 2, 29, 1, 0, 0, 0)
+                                        .format(
+                                                DateTimeFormatter.ofPattern(
+                                                        "yyyy-MM-dd' 'HH:mm:ss.SSS")),
```

Comment:
btw this could be extracted into constant since it's used multiple times

Reply:
ok